action/protocol/staking/protocol.go

```diff
@@ -442,7 +462,11 @@ func (p *Protocol) ReadState(ctx context.Context, sr protocol.StateReader, metho
 	case iotexapi.ReadStakingDataMethod_CANDIDATE_BY_ADDRESS:
 		resp, height, err = readStateCandidateByAddress(ctx, csr, r.GetCandidateByAddress())
 	case iotexapi.ReadStakingDataMethod_TOTAL_STAKING_AMOUNT:
-		resp, height, err = readStateTotalStakingAmount(ctx, csr, r.GetTotalStakingAmount())
+		resp, height, err = readStateTotalStakingAmountFromHeight(csr, r.GetTotalStakingAmount(), inputHeight)
```

Comment:
should be like below?
```
if p.archiveMode {
  resp, height, err = readStateTotalStakingAmountFromHeight(csr, r.GetTotalStakingAmount(), inputHeight)
} else {
  resp, height, err = readStateTotalStakingAmount(ctx, csr, r.GetTotalStakingAmount())
}
```

Reply:
I still don't kown how to get balance with height after Greenland?I think only lastest balance is saved after Greenland

Reply:
in archive mode, SR is a `historyStateReader`, calling State() will call sf.StateAtHeight(), which is the value at the height you pass to `historyStateReader`